jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -58,6 +61,9 @@
 // TODO: Add tests once containerize() is added.
 public class JibContainerBuilder {
 
+  @VisibleForTesting
+  static Supplier<ExecutorService> executorServiceFactory = Executors::newCachedThreadPool;
```

Comment:
Personally, I try avoiding swapping out a field for testing particularly for a static field (or it is just me thinking this way?), but I guess there is no other better way.

Reply:
Just realized a better approach.